plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/Fabric8WorkspaceEnvironmentProvider.java

```diff
@@ -153,21 +174,33 @@ public Config getWorkspacesOpenshiftConfig() throws OpenShiftException {
         | ConflictException
         | BadRequestException
         | IOException e) {
-      throw new RuntimeException(
-          "Cound not retrieve OSO token from Keycloak token: " + keycloakToken, e);
+      throw new RuntimeException("Cound not retrieve OSO token from Keycloak token", e);
     }
   }
 
-  public String getOpenShiftTokenForUser() throws OpenShiftException {
-    String keycloakToken = EnvironmentContext.getCurrent().getSubject().getToken();
+  public String getOpenShiftTokenForUser(Subject subject) throws OpenShiftException {
+
+    checkSubject(subject);
+
+    String keycloakToken = subject.getToken();
     if (keycloakToken == null) {
-      throw new OpenShiftException("User Openshift token is needed but there is no current user");
+      throw new OpenShiftException(
```

Comment:
maybe add a inner method to display subject 
```
subject.getUserName()
 +              + "("
 +              + subject.getUserId()
 +              + ")",
```
as it is used several times in this class